Gradebookidentityaddition

dtracers opened the pull request:
Adds Identity code to the gradebook

depends on  #482

dtracers committed:
adding some aasumptions


Former-commit-id: 78459d03341ef25af4bd8b49ac1a0dc5fe2ca966

dtracers committed:
Merge remote-tracking branch 'refs/remotes/origin/gradebookMergeFixings' into gradeBookIdentityAddition


Former-commit-id: 7e143e5b414547c1b51437286c5ccf4e6b0e81e7

dtracers committed:
added a userId to getting grades by their student value


Former-commit-id: 5b01442cc7a2aeca1ac60d49df89cf46e53559ca

dtracers committed:
changed how the course roster is returned.


Former-commit-id: 379fa3b9df9726bafc26917c144cf4661292adbe

dtracers committed:
addressed maven issues


Former-commit-id: 51cc7b30ecc38e0d86d7fa1c68dfbb08d931484e

dtracers committed:
Fixed issue with auth id and userid being the same

dtracers commented:
I think this branch needs more unit tests for getting a grade of the different types

dtracers committed:
Merge remote-tracking branch 'refs/remotes/origin/gradebook' into gradeBookIdentityAddition

dtracers committed:
Merge remote-tracking branch 'refs/remotes/origin/gradebook' into gradeBookIdentityAddition

# Conflicts:

#	CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/handlers/subhandlers/GradingRequestHandler.java

areftw commented on CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/Institution.java:
```diff
@@ -551,7 +554,7 @@ void insertSubmission(final String userId, String authId, String problemId, Stri
      *         Thrown if grades are not found in the database.
      */
-    List<ProtoGrade> getAllAssignmentGradesStudent(final String courseId, final String authId)
+    List<ProtoGrade> getAllAssignmentGradesStudent(final String courseId, final String authId, String userId)
```
make userId a final

areftw commented on CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/Institution.java:
```diff
@@ -593,7 +599,7 @@ void insertSubmission(final String userId, String authId, String problemId, Stri
      *         Thrown if a grade is not found in the database matching the requested parameters.
      */
-    ProtoGrade getGrade(final String authId, final ProtoGrade gradeData)
+    ProtoGrade getGrade(final String userId, final String authId, final ProtoGrade gradeData)
```
please put userId after authId like in the previous method

dtracers replied:
actually I believe the userId always goes before the authId

areftw commented on CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/Institution.java:
```diff
@@ -607,6 +613,19 @@ ProtoGrade getGrade(final String authId, final ProtoGrade gradeData)
+     * @throws DatabaseAccessException Thrown if the user does not exist or there are problems getting the username.
+     * @throws AuthenticationException Thrown if the user does not have the authentication to get the username.
+     */
+    String getUserNameForIdentity(final String userId, final String authId, final String courseId)
```
In getAllAssignmentGradesStudent you put userId last
pick an order and stick to it so it is constant for all methods

areftw commented on CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/GradeManager.java:
```diff
@@ -99,6 +99,8 @@ private GradeManager() {
      * @param grade
      *         The ProtoObject representing the grade to be added.
+     *         Assumptions:
+     *              The userId is the hashed userId.
```
wait waht?? what was the original userId then?

areftw commented on CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/MongoInstitution.java:
```diff
@@ -597,18 +598,29 @@ public void addGrade(final String authId, final ProtoGrade grade) throws Authent
     @Override
-    public ProtoGrade getGrade(final String authId, final ProtoGrade gradeData) throws AuthenticationException, DatabaseAccessException {
-        return GradeManager.getGrade(auth, database, authId, authId, gradeData);
+    public ProtoGrade getGrade(final String userId, final String authId, final ProtoGrade gradeData)
```
put user id after auithid plzzzz

areftw commented on CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/MongoInstitution.java:
```diff
@@ -597,18 +598,29 @@ public void addGrade(final String authId, final ProtoGrade grade) throws Authent
 
     @Override
-    public List<String> getCourseRoster(final String authId, final String courseId)
+    public String getUserNameForIdentity(final String userId, final String authId, final String courseId)
```
put userId at the end or change the others to userId in the beginning

areftw commented on CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/handlers/subhandlers/GradingRequestHandler.java:
```diff
@@ -56,14 +56,15 @@ private GradingRequestHandler() {
      * @param institution The database interface.
      * @param request The request being sent.
-     * @param authId The Id of the user who sent the request.
+     * @param authId The Id of the user who sent the request used for authentication purposes.
```
make sure you update the doc of @param authId for all other methods that contain authId

areftw commented on CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/handlers/subhandlers/GradingRequestHandler.java:
```diff
@@ -76,7 +77,7 @@ private GradingRequestHandler() {
             returnList = institution.getAllAssignmentGradesInstructor(request.getItemId(COURSE_INDEX), authId);
         } else if (student && allGrades) {
-            returnList = institution.getAllAssignmentGradesStudent(request.getItemId(COURSE_INDEX), authId);
+            returnList = institution.getAllAssignmentGradesStudent(request.getItemId(COURSE_INDEX), authId, userId);
```
change the method that you use here to take userId before authId

dtracers committed:
Addressed comments

dtracers committed:
Added the hashing of the grade to the database.

The grades were not correctly being hashed on the database server but
now they are correctly hashed on the database.

dtracers committed:
Added string versions of the hashing algorithm
